src/hivpy/demographics.py

```diff
@@ -1,19 +1,149 @@
+import warnings
+
 import numpy as np
 from pandas.api.types import CategoricalDtype
+from scipy.interpolate import interp1d
 
 SexType = CategoricalDtype(["female", "male"])
 
 
 # Default values
 # Can wrap those in something that ensures they have a description?
 FEMALE_RATIO = 0.52
+USE_STEPWISE_AGES = False
+INC_CAT = 1
+
+
+class StepwiseAgeDistribution:
+    # stepwise distributions
+    stepwise_model1 = np.array([0.18, 0.165, 0.144, 0.114, 0.09, 0.08, 0.068,
+                                0.047, 0.036, 0.027, 0.021, 0.016, 0.012])
+    stepwise_model2 = np.array([0.15, 0.13, 0.12, 0.11, 0.1, 0.09, 0.08, 0.065,
+                                0.048, 0.04, 0.03, 0.021, 0.016])
+    stepwise_model3 = np.array([0.128, 0.119, 0.113, 0.104, 0.097, 0.09, 0.081,
+                                0.074, 0.06, 0.05, 0.038, 0.026, 0.02])
+    stepwise_boundaries = np.array([-68, -55, -45, -35, -25, -15, -5, 5, 15,
+                                    25, 35, 45, 55, 65])
+
+    def __init__(self, ages, probabilities):
```

Comment:
Could add a check here that
```suggestion
    def __init__(self, ages, probabilities):
        assert len(ages) == len(probabilities) + 1
```

to ensure the generation works correctly later.